src/configurationProvider.ts

```diff
@@ -101,11 +101,11 @@ export class JavaDebugConfigurationProvider implements vscode.DebugConfiguration
              * If no launch.json exists in the current workspace folder
              * delegate to provideDebugConfigurations api to generate the initial launch.json configurations
              */
-            if (Object.keys(config).length === 0 && folder !== undefined) {
+            if (!config.request && folder !== undefined) {
```

Comment:
should write a common function to detect whether it's an empty config. Empty check should filter out the auto filled field `noDebug`.